### ISSUE TITLE
Swap most uses of allow in lints to expect

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#15059]
## What problem does this solve or what need does it fill?

With the advent of [Rust 1.81](https://blog.rust-lang.org/2024/09/05/Rust-1.81.0.html), we can now distinguish between lints that we haven't gotten around to fixing and those where the lint is wrong.

## What solution would you like?

Most localized lints in Bevy should be `expect`: the only exception that comes to mind is `missing_docs`.

Leave the `Cargo.toml` lints set as allow, as funny as it would be to set a crate-level `expect(type_complexity)`.

**Comment by alice-i-cecile** [comment]
I'd also like to turn on the new `allow_attributes_without_reason` lint as part of this.

**Comment by BD103** [comment]
I'll take a stab at this!

**Comment by BD103** [comment]
If someone wants to work on this, I got a lot of progress done in #15118. Feel free to adopt it and follow [my advice here](https://github.com/bevyengine/bevy/pull/15118?scrlybrkr=9612f9dd#issuecomment-2348935105) for finishing it.

**Comment by alice-i-cecile** [comment]
My suggestion for those tackling this is to handle this one crate at a time: that will ensure it's easier to complete and review smaller bits of work.

**Comment by richchurcher** [comment]
Copying BD103's instructions/workflow for ease of reference. Also note that crate-level `expect`/`missing_docs` should be fixed in an upcoming Rust version.

> I'm going to put this PR up for adoption. I no longer have the energy to work on it, and there are still a _lot_ of crates left. If you want to adopt this, my workflow looked like this:
> 
> 1. Run `cargo clippy --workspace --message-format=short`
> 2. `cd` into the first crate that raises warnings.
> 3. Run `cargo watch -c -x clippy` until warnings stop appearing.
> 4. Run `cargo clippy --no-default-features` and `cargo clippy --all-features` for good measure.
> 5. Repeat step 1 until there are no more warning.

**Comment by richchurcher** [comment]
Keeping a list here for my own reference: these were not covered by the previous PR.

- [ ] bevy_core_pipeline
- [ ] bevy_dev_tools
- [ ] bevy_dylib
- [ ] bevy_gilrs
- [ ] bevy_gizmos
- [ ] bevy_gltf
- [ ] ~bevy_input~ _looks pretty well-covered_
- [ ] bevy_internal
- [ ] bevy_pbr
- [ ] bevy_picking
- [ ] bevy_render
- [ ] bevy_sprite
- [ ] bevy_state
- [ ] bevy_text
- [ ] bevy_ui
- [ ] bevy_window
- [ ] bevy_winit

**Comment by richchurcher** [comment]
Just tracking things for myself... it looks like stable Rust 1.83 is 62 days away, so I have a reminder to revisit this end of November when we can rely on `missing_docs` working as expected.

**Comment by richchurcher** [comment]
Looking into this again...

**Comment by alice-i-cecile** [comment]
@bas-ie, are we fully done with this at this point?

**Comment by bas-ie** [comment]
I don't think so. I haven't looked specifically lately, but I think we just got the instances that were in BD's original PR and a few extras. I'll go hunting for more 👀
